diff --git a/samples/BUILD.gn b/samples/BUILD.gn
index f2c8014b0..89901d027 100644
--- a/samples/BUILD.gn
+++ b/samples/BUILD.gn
@@ -32,6 +32,38 @@ config("pdfium_samples_config") {
   }
 }
 
+shared_library("pdfium_shared") {
+  testonly = true
+  sources = [
+    "pdfium_test_lib.cc",
+    "pdfium_test_dump_helper.cc",
+    "pdfium_test_dump_helper.h",
+    "pdfium_test_event_helper.cc",
+    "pdfium_test_event_helper.h",
+    "pdfium_test_write_helper.cc",
+    "pdfium_test_write_helper.h",
+  ]
+  deps = [
+    "../:pdfium",
+    "../testing/:test_support",
+    "../testing/image_diff",
+    "//build/win:default_exe_manifest",
+  ]
+  configs += [ ":pdfium_samples_config" ]
+
+  if (pdf_enable_v8) {
+    deps += [
+      "//v8:v8_headers",
+      "//v8:v8_libplatform",
+    ]
+    include_dirs = [ "//v8" ]
+    configs += [ "//v8:external_startup_data" ]
+  }
+  if (pdf_use_skia || pdf_use_skia_paths) {
+    deps += [ "//skia" ]
+  }
+}
+
 executable("pdfium_test") {
   testonly = true
   sources = [
