diff --git a/core/fxge/BUILD.gn b/core/fxge/BUILD.gn
index 6c9ed9a11..fd480daa8 100644
--- a/core/fxge/BUILD.gn
+++ b/core/fxge/BUILD.gn
@@ -157,7 +157,7 @@ jumbo_source_set("fxge") {
     ]
   }
 
-  if (is_linux) {
+  if (is_linux || current_os == "wasm") {
     sources += [ "fx_ge_linux.cpp" ]
   }
 
diff --git a/testing/utils/path_service.cpp b/testing/utils/path_service.cpp
index 5e1ce3960..91bd60e85 100644
--- a/testing/utils/path_service.cpp
+++ b/testing/utils/path_service.cpp
@@ -10,7 +10,8 @@
 #include <mach-o/dyld.h>
 #include <sys/stat.h>
 #else  // Linux
-#include <linux/limits.h>
+//#include <linux/limits.h>
+#define PATH_MAX 4096
 #include <sys/stat.h>
 #include <unistd.h>
 #endif  // _WIN32
diff --git a/third_party/BUILD.gn b/third_party/BUILD.gn
index 51fd0cac4..4d503ef40 100644
--- a/third_party/BUILD.gn
+++ b/third_party/BUILD.gn
@@ -446,6 +446,7 @@ source_set("fx_libopenjpeg") {
     "libopenjpeg20/tgt.c",
     "libopenjpeg20/thread.c",
   ]
+  defines = [ "_GNU_SOURCE" ]
 }
 
 config("system_libpng_config") {
diff --git a/third_party/base/allocator/partition_allocator/spin_lock.cc b/third_party/base/allocator/partition_allocator/spin_lock.cc
index 42055836b..1aa96a72a 100644
--- a/third_party/base/allocator/partition_allocator/spin_lock.cc
+++ b/third_party/base/allocator/partition_allocator/spin_lock.cc
@@ -51,7 +51,7 @@
 #endif  // ARCH
 
 #ifndef YIELD_PROCESSOR
-#warning "Processor yield not supported on this architecture."
+//#warning "Processor yield not supported on this architecture."
 #define YIELD_PROCESSOR ((void)0)
 #endif
 
diff --git a/third_party/yasm/yasm_assemble.gni b/third_party/yasm/yasm_assemble.gni
index 73d554856..67f40b24a 100644
--- a/third_party/yasm/yasm_assemble.gni
+++ b/third_party/yasm/yasm_assemble.gni
@@ -123,6 +123,7 @@ template("yasm_assemble") {
                          root_build_dir) ]
 
     # Deps.
+    print(invoker)
     deps = [
       yasm_label,
     ]
