From 92643e87e568ed961dd73dc107874280963baeb0 Mon Sep 17 00:00:00 2001
From: Andrew de los Reyes <adlr@chromium.org>
Date: Thu, 28 Nov 2019 14:57:32 -0800
Subject: [PATCH 2/3] Support for fallback font selection (needed for emoji)

---
 src/ports/SkFontMgr_custom.cpp | 16 +++++++++++++++-
 1 file changed, 15 insertions(+), 1 deletion(-)

diff --git a/src/ports/SkFontMgr_custom.cpp b/src/ports/SkFontMgr_custom.cpp
index 6ee4e5454f..a3db9801a9 100644
--- a/src/ports/SkFontMgr_custom.cpp
+++ b/src/ports/SkFontMgr_custom.cpp
@@ -206,10 +206,24 @@ SkTypeface* SkFontMgr_Custom::onMatchFamilyStyle(const char familyName[],
 }
 
 SkTypeface* SkFontMgr_Custom::onMatchFamilyStyleCharacter(const char familyName[],
-                                                          const SkFontStyle&,
+                                                          const SkFontStyle& style,
                                                           const char* bcp47[], int bcp47Count,
                                                           SkUnichar character) const
 {
+    // Try each family and see if they have support
+    int cnt = countFamilies();
+    for (int i = 0; i < cnt; i++) {
+        SkString family;
+        getFamilyName(i, &family);
+        if (family.isEmpty())
+            continue;
+        SkTypeface* face = matchFamilyStyle(family.c_str(), style);
+        if (!face)
+            continue;
+        if (!face->unicharToGlyph(character))
+            continue;
+        return face;
+    }
     return nullptr;
 }
 
-- 
2.11.0

